jumanji/environments/routing/cmst/reward.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from jumanji.environments.routing.cmst.constants import (
+    INVALID_CHOICE,
+    INVALID_TIE_BREAK,
+)
+from jumanji.environments.routing.cmst.types import State
+
+
+class RewardFn(abc.ABC):
+    """Abstract class for `Connector` rewards."""
+
+    @abc.abstractmethod
+    def __call__(
+        self, state: State, actions: chex.Array, nodes_to_connect: chex.Array
+    ) -> chex.Array:
+        """The reward function used in the CMST environment.
+
+        Args:
+            state: Environment state
+            actions: Actions taken by all the agents to reach this state.
+            nodes_to_connect: Array containing the nodes each agent needs to connect.
+        Returns:
+            reward
+        """
+
+
+class DefaultRewardFn(RewardFn):
```

Comment:
Could we give it a more explicit name? The fact that it is the default used by the environment does not characterise the reward function in itself.

Reply:
I have named it DenseRewardFn